### PR TITLE
Add api-extractor to help reviewing API changes

### PR DESCRIPTION
**Pull request by jeremymeng**
## Description

This PR adds `@microsoft/api-extractor` dev dependency to help review API changes and identify issues of missing exports in public API surface.

### CODE REVIEW

File: review/rhea-promise.api.md [hunk 1]
```diff
@@ -0,0 +1,589 @@
+## API Report File for "rhea-promise"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+/// <reference types="node" />
+
+import { AmqpError } from 'rhea';
+import { Connection as Connection_2 } from 'rhea';
```
Comment (HarshaNalluru) [inline review/rhea-promise.api.md:10]: is it possible to avoid these?

File: review/rhea-promise.api.md [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    message?: Message;
+    receiver?: Receiver;
+    reconnecting?: boolean;
+    sender?: Sender;
+    session?: Session;
+}
+
+// @public (undocumented)
+export namespace EventContext {
+    // Warning: (ae-forgotten-export) The symbol "Link" needs to be exported by the entry point index.d.ts
```
Comment (HarshaNalluru) [inline review/rhea-promise.api.md:358]: should we export?

File: review/rhea-promise.api.md [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+    // (undocumented)
+    on(event: SessionEvents, listener: OnAmqpEvent): this;
+}
+
+// @public
+export class Session extends Entity {
+    constructor(connection: Connection, session: Session_2);
+    // (undocumented)
+    begin(): void;
+    // Warning: (ae-forgotten-export) The symbol "SessionCloseOptions" needs to be exported by the entry point index.d.ts
```
Comment (HarshaNalluru) [inline review/rhea-promise.api.md:531]: this

File: review/rhea-promise.api.md [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+export interface Connection {
+    // (undocumented)
+    on(event: ConnectionEvents, listener: OnAmqpEvent): this;
+}
+
+// Warning: (ae-forgotten-export) The symbol "Entity" needs to be exported by the entry point index.d.ts
+//
+// @public
+export class Connection extends Entity {
+    // Warning: (ae-forgotten-export) The symbol "CreatedRheaConnectionOptions" needs to be exported by the entry point index.d.ts
```
Comment (HarshaNalluru) [inline review/rhea-promise.api.md:204]: Thanks for this PR. I think not having an api document caused all these missed exports.

**Review by HarshaNalluru — APPROVED** [review]
Looks good. But we should consider exporting the missed pieces and publish a minor version for completion. (maybe as a separate PR)

File: review/rhea-promise.api.md (hunk 1 above)
Comment (jeremymeng) [reply review/rhea-promise.api.md:10]: It's a known api-extractor issue when there are types with the same names. I can live with it even it is not great.

**Pull request merged**